backend/zeno_backend/database/update.py

```diff
@@ -226,11 +227,47 @@ def project_metrics(project_config: Project):
         project_config (Project): the configuration of the project.
     """
     with Database() as db:
-        db.execute(
-            "DELETE FROM metrics WHERE project_uuid = %s;",
+        metric_results = db.execute_return(
+            "SELECT id, name, type, columns FROM metrics WHERE project_uuid = %s;",
             [project_config.uuid],
         )
-        for metric in project_config.metrics:
+        metrics = list(
+            map(
+                lambda metric: Metric(
+                    id=metric[0],
+                    name=metric[1],
+                    type=metric[2],
+                    columns=metric[3],
+                ),
+                metric_results,
+            )
+        )
+        new_metrics = project_config.metrics
+
+        for metric in metrics:
+            # check if the metric matches one to be added
+            index = next(
```

Comment:
nit: this is a bit unreadable, I don't really know what it's doing. Is there a simpler way to do this logic? If not and it works now worries.

Reply:
Hmm could also loop but less pretty I'd say....

It's just getting the next element that matches the condition. If not element matches the condition, returns None.